### PR TITLE
Add support for building VM images

### PR DESCRIPTION
**Pull request by obierlaire**
Related to https://github.com/elastic/ansible-elastic-cloud-enterprise/issues/56
Related to https://github.com/elastic/cloud/issues/51245

This is enabling building AMIs by reorganize/rename the roles/tasks and adding a new task (`ami`). In case of AMI there are some specificities (using tag `ami`)
- no reboot
- no install of ECE (although, you still can, using tags `ami` AND `bootstrap`)
- no disk setup (handled by cloud-init script)
- docker disabled (handled by cloud-init script, disk should be mounted first)

Tags can be used:
- no tag: prerequistes + install ECE as before
- system+ece+boostrap : same as above
- sytsem+ami: install prerequistes + AMI specific files
- bootstrap: install ECE only (assuming prerequistes are met)
- sytsem+ami+boostrap: install prerequistes + AMI specific files + install ECE

With no tags, this will still install prerequistes and boostrap ECE as before.

### CODE REVIEW

**Comment by vaubarth** [comment]
Can you also adapt the README to reflect the changes, especially for the supported tags and a section about the ami building in general.
Also `CONTRIBUTING.md` needs changes to reflect the new path structure and to reflect that ami buidling should be supported for new OSs

**Comment by vaubarth** [comment]
One more general thing, if I am not mistaken this is not only relevant for AMIs but for all kinds of prebuild images, right?
If so then we should reflect that in the naming and change `ami` to maybe `prebuild` or something

**Comment by obierlaire** [comment]
Thanks for the reviews. @vaubarth I updated the docs to reflect those changes, and changed `ami` -> `vmimage` as discussed

File: templates/format-drives.j2 [hunk 1]
```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
```
Comment (vaubarth) [inline templates/format-drives.j2]: I am unclear where this files gets copied over, am I missing something or is there a missing piece to actually put this template in place?

File: templates/format-drives.j2 [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+lvcreate -n swap -L $(grep MemTotal /proc/meminfo | awk -v MAXMEM=${SWAP_MAX_SIZE} '{ mem=int($2/(2*1024)); if(mem>MAXMEM) mem=MAXMEM; print mem; }')m ${LVG}
+lvcreate -n data -l 100%FREE ${LVG}
+
+echo "Formatting LVM volumes..."
+mkswap /dev/${LVG}/swap
+mkfs.xfs -K /dev/${LVG}/data
+
+DATA_MOUNT_OPTIONS="noatime,defaults,pquota,prjquota"
+
+{% if systemd_os %}
```
Comment (vaubarth) [inline templates/format-drives.j2]: This probably clears up when I understand how this template is copied, but where is `systemd_os`  set?

File: templates/format-drives.j2 (hunk 1 above)
Comment (obierlaire) [reply templates/format-drives.j2]: @vaubarth I'm sorry I forgot to add the renamed folder `vmimage`, I just added back and you will see easily where it's copied (`tasks/vmimage/main.yml`)

**Pull request merged**